daemon/debug.go

```diff
@@ -0,0 +1,89 @@
+package daemon
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// SupportDump returns a tar with details often required to debug user issues.
+// The format of this tar is not stable and is intended for human consumption.
+func (daemon *Daemon) SupportDump(ctx context.Context) (io.Reader, error) {
```

Comment:
Perhaps this should take a list of functional arguments that collect the actual data; that way it could be extended more easily? (ie adding custom "collectors")